hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -140,11 +140,21 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
     Map<String, List<SmallFile>> partitionSmallFilesMap =
         getSmallFilesForPartitions(new ArrayList<String>(partitionPaths), context);
 
+    // get the in pending clustering fileId for each partition path
+    Map<String, List<String>>  partitionPathToInPendingClusteringFileId =
```

Comment:
looks like Map<String, Set<String>> makes more sense because we only do contains check?